addons/mail/static/src/core/common/store_service.js

```diff
@@ -478,6 +478,19 @@ export class Store extends BaseStore {
         return chat;
     }
 
+    fillPartnersMentionToken(postData) {
+        postData.partner_ids_mention_token ||= {};
+        Object.assign(
+            postData.partner_ids_mention_token,
+            Object.fromEntries(
+                postData.partner_ids
+                    .map((pid) => this["res.partner"].get(pid))
+                    .filter((partner) => partner && partner.mention_token)
+                    .map((partner) => [partner.id, partner.mention_token])
+            )
+        );
+    }
+
```

Comment:
```suggestion
    fillPartnersMentionToken(postData) {
        postData.partner_ids_mention_token ||= {};
        for (const pid of postData.partner_ids) {
            const partner = this["res.partner"].get(pid);
            if (partner?.mention_token) {
                postData.partner_ids_mention_token[pid] = partner.mention_token;
            }
    }
```
:pick: I feel like this could be done in a more readable, less nested way.